documentation updates (#951)
elastic/apm-agent-go: docs/api.asciidoc

[[api]]
== API Documentation

> Below are the most commonly used parts of the API.

Author: ```suggestion
This section describes the most commonly used parts of the API.
```

[[transaction-ensureparent]]
==== `func (*Transaction) EnsureParent() SpanID`

EnsureParent returns the transaction's parent span ID--generating and recording one if
it did not previously have one.

> EnsureParent generates and records the transaction parent span ID if it did not previously have one.
> 

Author: ```suggestion
```

[[context-api]]
=== Context

Built-in instrumentation will typically provide some context when reporting transaction and error events,

> like the URL and remote address for an HTTP request.

Author: Reverted

> Built-in instrumentation will typically provide some context when reporting transaction and error events,
> like the URL and remote address for an HTTP request.
> You can add additional context with custom context and tags.

Author: ```suggestion
When reporting transactions and errors you can provide context to describe
those events. Built-in instrumentation will typically provide some context,
e.g. the URL and remote address for an HTTP request. You can also provide
custom context and tags.
```
